General refactoring and consolidation

waldyrious opened the pull request:
This PR implements various cleanup changes to consolidate and clean up the repository contents and structure after the initial seeding in #1, #2, #3 and #4. Namely, the following changes have been performed:

- Clean up git tracking of package lock files;
- Refactor and standardize syntax and code formatting across all files, including harmonization of the syntax of Axios requests, and simplification of the transaction sample code;
- Trim down samples' dependencies (removing over a hundred dependent packages);
- Add main README file;
- Improve documentation in sample projects' names, descriptions, and READMEs;
- Reorganize and rename directories to provide a clearer and more self-consistent naming scheme for the samples, and facilitate navigation.

waldyrious committed:
Update package lock definitions

- Remove `yarn.lock` files and keep `package-lock.json` in all samples
- Remove ignore rule for `package-lock.json` files

waldyrious committed:
Cleanup and standardize file contents & formatting

waldyrious committed:
Standardize syntax of Axios requests

waldyrious committed:
Simplify transaction sample code

waldyrious committed:
Trim down samples' dependencies

- client credentials sample:  11 packages removed
- PAT sample:                  3 packages removed
- authorization code sample: 119 packages removed (!)
- transaction sample:          4 packages removed

waldyrious committed:
Improve documentation

- Add main README file
- Improve project names and descriptions
- Improve sample projects' READMEs

waldyrious committed:
Move samples to the root directory

Also rename the sample directories to a clear and more self-consistent naming scheme

inginheiiro reviewed (approved):
LGTM

Commit:
Use consistent and explicit names for source files

inginheiiro reviewed (changes requested):
fixed issues with **bearer** and names

waldyrious committed:
Standardize logging of API error messages

Co-authored-by: Paulo Cabral <paulo.cabral@uphold.com>

waldyrious committed:
Move error composition method to the index.js of the authorization code sample

This standardizes the code so that all the non-index files
contain only methods that wrap calls to the Uphold API

waldyrious committed:
Wrap the index files in try-catch blocks

This allows for cleaner failure when one of the API calls fails

waldyrious committed:
Display an informative error web page in OAuth sample

waldyrious committed:
Add checks for existence of the .env file

waldyrious committed:
Fix token type in request headers of OAuth examples

The access token type in the authorization header must be "Bearer",
with a capital B.
This change also makes the code of the OAuth examples
more consistent with that of the PAT example.

waldyrious committed:
Fix grant type in OAuth authorization code example

waldyrious committed:
Use an authenticated request to test access tokens in OAuth examples

The /assets endpoint is accessible without authentication,
so a call to it succeeds even with incorrect credentials,
which defeats the purpose of using it to test the authentication flows.

This commit switches it to a request to /me,
which does require authentication (and as a bonus, is less verbose).

waldyrious committed:
Consistently use 'access token' nomenclature for clarity

waldyrious committed:
Standardize logging of API call results in authentication examples
